File: backend/api/services/empresa_service.py
from ..models import empresa_model
from api.app import db

def insert_empresa(empresa):
    empresa_bd = empresa_model.EmpresaModel(nomeEmpresa=empresa.nomeEmpresa, dtRefSistema=empresa.dtRefSistema, dtImplantacao=empresa.dtImplantacao, endereco=empresa.endereco, cnpj=empresa.cnpj, cidade=empresa.cidade)
    db.session.add(empresa_bd)
    db.session.commit()
    return empresa_bd

def update_empresa(empresa):
    empresa_model.EmpresaModel.query.filter_by(idEmpresa=1).update(
            {"nomeEmpresa": empresa['nomeEmpresa'], "dtRefSistema": empresa['dtRefSistema'], "endereco": empresa['endereco'], "cnpj": empresa['cnpj'], "cidade": empresa['cidade']})

    db.session.commit()
    
    empresa_bd = get_empresa()
    return empresa_bd

def get_empresa():
    return empresa_model.EmpresaModel.query.filter_by(idEmpresa=1).first()

File: frontend/src/app/services/parametrizacao/empresa.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Empresa } from '../../models/empresa'


@Injectable({
  providedIn: 'root'
})
export class EmpresaService {
  private apiUrl = '';
  private Empresa:Empresa

  constructor(private http: HttpClient) {
    this.apiUrl  = environment.url
    this.Empresa = {
      idEmpresa:0,
      nomeEmpresa:'',
      dtRefSistema:'',
      dtImplantacao:'',
      endereco:'',
      cidade:'',
      cnpj:''
    }
   }

  getEmpresa(): Observable<Empresa> {
    let apiKey = sessionStorage.getItem('apiKey');

    const params = { apiKey: `${apiKey}` };
    return this.http.get<any>(`${this.apiUrl}/empresa`, { params });
  }
}
